relay-server/src/metrics_extraction/spans/mod.rs

```diff
@@ -303,6 +310,26 @@ fn sanitized_span_description(
     Some(sanitized)
 }
 
+/// Trims the given span description to ensure
+/// [`relay-metrics::AggregatorService`] doesn't drop it for being too long.
+///
+/// If the description is short, it remains unchanged. If it's long, this method
+/// truncates it to the maximum allowed length and sets the last character to
+/// `*`. The maximum length is determined by [`MaxChars::TagValue`].
+///
+/// Note [`relay-metrics::AggregatorService`] is configured independently and
+/// doesn't use this type; this method uses it for simplicity, and because at
+/// the time of writing this the values are the same. If the limit this type
+/// provides is longer, this method may be ineffective.
+fn truncate_span_description(description: String) -> String {
```

Comment:
Could it be better to take a mutable reference as an argument ? wrt allocations it won't matter if the caller has only a &str of course, but in the cases where it's an owned type then you can avoid allocating.

for the &str ones, is it possible to utilize a Cow maybe?

Reply:
If the caller has a `String`, my understanding is there are no additional allocations because it passes the ownership of the variable. If my understanding is correct, then `Cow` doesn't help here because there aren't additional allocations. Does this make sense, or am I missing something?